programs/whirlpool/src/instructions/v2/set_token_badge_authority.rs

```diff
@@ -0,0 +1,25 @@
+use anchor_lang::prelude::*;
+
+use crate::state::{WhirlpoolsConfig, WhirlpoolsConfigExtension};
+
+#[derive(Accounts)]
+pub struct SetTokenBadgeAuthority<'info> {
+    pub whirlpools_config: Box<Account<'info, WhirlpoolsConfig>>,
+
+    #[account(mut, has_one = whirlpools_config)]
+    pub whirlpools_config_extension: Account<'info, WhirlpoolsConfigExtension>,
+
+    #[account(address = whirlpools_config_extension.config_extension_authority)]
+    pub config_extension_authority: Signer<'info>,
+
+    /// CHECK: safe, the account that will be new authority can be arbitrary
+    pub new_token_badge_authority: UncheckedAccount<'info>,
+}
+
+/// Set the fee authority. Only the config extension authority has permission to invoke this instruction.
```

Comment:
For example, feeAuthority can set newFeeAuthority by signing the feeAuthority by themself. (set_fee_authority instruction can be executed by the signature of the current feeAuthority).

On the other hand, tokenBadgeAuthority cannot set newTokenBadgeAuthority by themself.
configExtensionAuthority need to do it.

This is designed to allow for the addition of new fields to WhirlpoolsConfigExtension easier.

WhirlpoolsConfigExtension has 512 bytes of reserved space that can be used for future additions.
Since the reserved area is initialized to 0, adding functionality without breaking the already created WhirlpoolsConfigExtension would involve adding a new setX instruction and having configExtensionAuthority execute it.
This is because newly adding authority is uninitialized and cannot set itself.

Rather than adding two sets of instructions, one for initialization and one for updating itself, one instruction can be added by using only configExtensionAuthority.

Reply:
But this is the instruction for setting the `badge authority` why is the comment talking about a `fee authority`?